<p>
    Having two branches in the same <code>@if ... @else if ... @else</code> structure with the same implementation is
    at best duplicate code, and at worst a coding error. If the same logic is truly needed for both instances, then they
    should be combined.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
@if $x == 0 {
  do-something();
} @else if $x == 1 {
  do-something-different();
} @else {
  do-something();   /* Noncompliant: Same implementation as in the @if block */
}
</pre>

<h2>Compliant Solution</h2>
<pre>
@if $x == 1 {
  do-something-different();
} @else {
  do-something();
}

OR

@if $x == 0 {
  do-something();
} @else if $x == 1 {
  do-something-different();
} @else {
  do-something-else();
}
</pre>
