<p> An empty mixin is one that doesn't contain any declaration. Empty mixins are useless and should be removed.</p>

<h2>Noncompliant Code Example</h2>
<pre>
@mixin my-mixin {}
</pre>

<h2>Compliant Solution</h2>
<pre>
@mixin my-mixin {
  width: 10px;
}
</pre>
