<p>An empty mixin is one that doesn't contain any declaration. Empty mixins are useless and should be removed.</p>

<h2>Noncompliant Code Example</h2>
<pre>
@mixin my-mixin {}
</pre>

<h2>Compliant Solution</h2>
<pre>
@mixin my-mixin {
  width: 10px;
}
</pre>

<h2>stylelint Related Rules</h2>
<ul>
  <li><a href="https://stylelint.io/user-guide/rules/block-no-empty/" target="_blank">block-no-empty</a></li>
</ul>
