<p>
    A chain of <code>@if</code> / <code>@else if</code> directives is evaluated from top to bottom. At most, only one
    branch will be executed: the first one with a condition that evaluates to true.
</p>
<p>
    Therefore, duplicating a condition automatically leads to dead code. Usually, this is due to a copy/paste error. At
    best, it's simply dead code and at worst, it's a bug that is likely to induce further bugs as the code is
    maintained, and obviously it could lead to unexpected behavior.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
@if $x == 0 {
  do-something();
} @else if x == 1 {
  do-something-else();
} @else if x == 0 {  /* Noncompliant: Same condition as in the @if directive */
  do-something-different();
}
</pre>

<h2>Compliant Solution</h2>
<pre>
@if $x == 0 {
  do-something();
} @else if x == 1 {
  do-something-else();
} @else if x == 2 {
  do-something-different();
}
</pre>
