<p> The complexity of a condition is defined by the number of <code>and</code> and <code>or</code> operators.</p>
<p> A single condition's complexity should not become too high to keep the code readable.</p>

<h2>Noncompliant Code Example</h2>
With a maximum allowed number of conditional operators of 3:
<pre>
@if $height < 10 px and $width < 20px and $depth < 5px or $is-box-valid {
  do-something();
}
</pre>

<h2>Compliant Solution</h2>
With a maximum allowed number of conditional operators of 3:
<pre>
$is-box-small-enough: $height < 10 px and $width < 20px and $depth < 5px;

@if $is-frame-small-enough or $is-box-valid {
  do-something();
}
</pre>
