<p>The box model is one of the most frequently misunderstood parts of CSS. "Box model" refers to the series of boxes that make up an element visually. This starts with the content, which is the inner-most box. Content is surrounded by padding, which in turn is surrounded by borders. The way these measurements interact, however, is a bit confusing. Consider the following:</p>
<pre>
.mybox {
  border: 1px solid black;
  padding: 5px;
  width: 100px;
}
</pre>
<p>A new developer might assume that the width of an element with the <code>mybox</code> class would be 100 pixels. In fact, the width is 112 pixels because width refers to the content box and padding and borders are added on top of that. Frequently when developers include this combination of properties, it is an error because they are expecting different behavior.</p>
<p>You can force most browsers to obey <code>width</code> and <code>height</code> as the full size of an element by using the <code>box-sizing</code> property and setting it to <code>border-box</code>, as in this example:</p>
<pre>
.mybox {
  box-sizing: border-box;
  border: 1px solid black;
  padding: 5px;
  width: 100px;
}
</pre>
<p>Now an element with a class of <code>mybox</code> will have an actual width of 100 pixels, and the padding and borders will exist inside of that area, leaving 88 pixels for content instead of 100 pixels.</p>
<p>This rule is aimed at eliminating unwanted box model sizing issues. As such, the rule raises an issue when it finds:
<ul>
  <li><code>width</code> being used with <code>border</code>, <code>border-left</code>, <code>border-right</code>, <code>border-width</code>, <code>border-left-width</code>, <code>border-right-width</code>, <code>padding</code>, <code>padding-left</code>, or <code>padding-right</code></li>
  <li><code>height</code> being used with <code>border</code>, <code>border-top</code>, <code>border-bottom</code>, <code>border-width</code>, <code>border-top-width</code>, <code>border-bottom-width</code>, <code>padding</code>, <code>padding-top</code>, or <code>padding-bottom</code></li>
</ul>
If the <code>box-sizing</code> property is specified, then the rule does not raise any issues for the above conditions as it assumes you know what you're doing.
</p>
<h2>Noncompliant Code Example</h2>
<pre>
/* Noncompliant: 'width' defined along with 'border' */
.mybox {
  border: 1px solid black;
  width: 100px;
}
</pre>

<h2>Compliant Solution</h2>
<pre>
/* Compliant: 'width' defined along with 'border' but 'box-sizing' is defined as well */
.mybox {
  box-sizing: border-box;
  border: 1px solid black;
  width: 100px;
}

/* Compliant: 'width' defined along with 'border-width' but 'border-width' is equals to zero */
.mybox {
  border-width: 0;
  width: 100px;
}
</pre>

<h2>See</h2>
<ul>
  <li><a href="https://github.com/stubbornella/csslint/wiki/Beware-of-box-model-size">CSSLint rule documentation</a></li>
</ul>
