<p>
    When using <code>@font-face</code> to declare multiple font types for cross browser compatibility, you can see 404's
    in old versions of IE due to a bug in the way that IE parses the font declarations.
<h2>Noncompliant Code Example</h2>
<p>The following piece of code causes a 404 in IE 6, 7, and 8.</p>
<pre>
@font-face {
  font-family: 'MyFontFamily';
  src: url('myfont-webfont.eot') format('embedded-opentype'),
       url('myfont-webfont.woff') format('woff'),
       url('myfont-webfont.ttf')  format('truetype'),
       url('myfont-webfont.svg#svgFontName') format('svg');
}
</pre>
<h2>Compliant Solution</h2>
<p>The fix is to add a question mark after the first font URL, so IE sees the rest of the property value as a query
    string.</p>
<pre>
@font-face {
  font-family: 'MyFontFamily';
  src: url('myfont-webfont.eot?#iefix') format('embedded-opentype'),
       url('myfont-webfont.woff') format('woff'),
       url('myfont-webfont.ttf')  format('truetype'),
       url('myfont-webfont.svg#svgFontName') format('svg');
}
</pre>
</p>
<h2>See</h2>
<ul>
    <li><a href="https://github.com/stubbornella/csslint/wiki/Bulletproof-font-face">CSSLint rule documentation</a></li>
</ul>

<p>
    This rule is deprecated, use {rule:css:font-face-browser-compatibility} instead.
</p>
