<p>Even though CSS syntax is case-insensitive, sharing some style conventions is a key point to make it possible for a team to efficiently collaborate.</p>
<p>This rule allows to check that all properties and functions are written in lowercase.</p>
<h2>Noncompliant Code Example</h2>
<pre>
.mybox {
  color: blue;
  COLOR: blue; /* Noncompliant */
  Color: blue; /* Noncompliant */;
}

.mybox {
  color: rgb(255,255,255;
  color: RGB(255,255,255; /* Noncompliant */
  color: Rgb(255,255,255; /* Noncompliant */;
}
</pre>
<h2>Compliant Solution</h2>
<pre>
.mybox {
  color: blue;
}

.mybox {
  color: rgb(255,255,255;
}
</pre>