<p>
    The case-insensitive flag is not currently supported by all browsers (Internet Explorer and Edge for instance). Thus, it
    should not be used.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
[frame=hsides i] {
  border-style: solid none;
}
</pre>

<h2>See</h2>
<ul>
    <li><a href="https://drafts.csswg.org/selectors-4/#attribute-case">W3C Selectors Specifications Level 4</a></li>
    <li><a href="http://caniuse.com/#feat=css-case-insensitive">Can I use</a></li>
</ul>
