<p>Even though CSS syntax is case-insensitive (except for variables), sharing some style conventions is a key point to make it possible for a team to efficiently collaborate.</p>
<p>This rule allows to check that all properties, functions and variables are written in lowercase.</p>
<h2>Noncompliant Code Example</h2>
<pre>
.mybox {
  COLOR: blue; /* Noncompliant */
}

.mybox {
  color: RGB(255,255,255); /* Noncompliant */
}

:root {
  --MyVar: red; /* Noncompliant */;
}
</pre>
<h2>Compliant Solution</h2>
<pre>
.mybox {
  color: blue;
}

.mybox {
  color: rgb(255,255,255);
}

:root {
  --myvar: red;
}
</pre>