<p>The <code>@charset</code> rule is invalid if it is not the first element in the style sheet or if it is preceded by any character including whitespace.</p>

<h2>Noncompliant Code Example</h2>
<pre>
    @charset "UTF-8" /* Noncompliant: preceding whitespaces */
</pre>

<h2>See</h2>
<ul>
    <li><a href="https://www.w3.org/TR/css-syntax-3/#charset-rule">W3C Specifications</a></li>
    <li><a href="https://developer.mozilla.org/en-US/docs/Web/CSS/@charset">Mozilla Developer Network @charset Documentation</a></li>
</ul>
