<p>Experimental CSS properties are typically implemented using vendor prefixes until the final behavior has been established and agreed upon. Most CSS3 properties have vendor-prefixed equivalents for multiple vendors, including Firefox (<code>-moz</code>), Safari/Chrome (<code>-webkit</code>), Opera (<code>-o</code>), and Internet Explorer (<code>-ms</code>). It's easy to forget to include the vendor prefixed version of a property when there are so many to keep track of.</p>
<p>The following properties have multiple vendor-prefixed versions:
<ul>
  <li><code>animation</code></li>
  <li><code>animation-delay</code></li>
  <li><code>animation-direction</code></li>
  <li><code>animation-duration</code></li>
  <li><code>animation-fill-mode</code></li>
  <li><code>animation-iteration-count</code></li>
  <li><code>animation-name</code></li>
  <li><code>animation-play-state</code></li>
  <li><code>animation-timing-function</code></li>
  <li><code>appearance</code></li>
  <li><code>border-end</code></li>
  <li><code>border-end-color</code></li>
  <li><code>border-end-style</code></li>
  <li><code>border-end-width</code></li>
  <li><code>border-image</code></li>
  <li><code>border-radius</code></li>
  <li><code>border-start</code></li>
  <li><code>border-start-color</code></li>
  <li><code>border-start-style</code></li>
  <li><code>border-start-width</code></li>
  <li><code>box-align</code></li>
  <li><code>box-direction</code></li>
  <li><code>box-flex</code></li>
  <li><code>box-lines</code></li>
  <li><code>box-ordinal-group</code></li>
  <li><code>box-orient</code></li>
  <li><code>box-pack</code></li>
  <li><code>box-sizing</code></li>
  <li><code>box-shadow</code></li>
  <li><code>column-count</code></li>
  <li><code>column-gap</code></li>
  <li><code>column-rule</code></li>
  <li><code>column-rule-color</code></li>
  <li><code>column-rule-style</code></li>
  <li><code>column-rule-width</code></li>
  <li><code>column-width</code></li>
  <li><code>hyphens</code></li>
  <li><code>line-break</code></li>
  <li><code>margin-end</code></li>
  <li><code>margin-start</code></li>
  <li><code>marquee-speed</code></li>
  <li><code>marquee-style</code></li>
  <li><code>padding-end</code></li>
  <li><code>padding-start</code></li>
  <li><code>tab-size</code></li>
  <li><code>text-size-adjust</code></li>
  <li><code>transform</code></li>
  <li><code>transform-origin</code></li>
  <li><code>transition</code></li>
  <li><code>transition-delay</code></li>
  <li><code>transition-duration</code></li>
  <li><code>transition-property</code></li>
  <li><code>transition-timing-function</code></li>
  <li><code>user-modify</code></li>
  <li><code>user-select</code></li>
  <li><code>word-break</code></li>
  <li><code>writing-mode</code></li>
</ul>
</p>
<p>
This rule is intended to raise an issue when a vendor-prefixed property is missing. Note that only the following main vendors are checked by this rule:
<ul>
  <li>Firefox (<code>-moz</code>)</li>
  <li>Safari/Chrome (<code>-webkit</code>)</li>
  <li>Opera (<code>-o</code>)</li>
  <li>Internet Explorer (<code>-ms</code>)</li>
</ul>
Konqueror (<code>khtml</code>) for example is not.
</p>
<h2>Noncompliant Code Example</h2>
<pre>
/* Noncompliant: Missing -moz, -ms, and -o */
.mybox {
  -webkit-transform: translate(50px, 100px);
}
</pre>
<h2>Compliant Solution</h2>
<pre>
.mybox {
  -webkit-transform: translate(50px, 100px);
  -moz-transform: translate(50px, 100px);
  -ms-transform: translate(50px, 100px);
  -o-transform: translate(50px, 100px);
}
</pre>
<h2>See</h2>
<ul>
  <li><a href="https://github.com/stubbornella/csslint/wiki/Require-compatible-vendor-prefixes">CSSLint rule documentation</a></li>
</ul>
