<p>Static filters are deprecated since Internet Explorer 9 and should not be used anymore.</p>
<h2>Noncompliant Code Example</h2>
<pre>
img {
  filter: alpha(opacity=40); /* Noncompliant: 'alpha' is an Internet Explorer static filter */
}
</pre>
<h2>See</h2>
<ul>
  <li><a href="https://msdn.microsoft.com/en-us/library/ms532847%28v=vs.85%29.aspx">Internet Explorer static filters</a></li>
</ul>