<p>System colors are deprecated and should not be used anymore.</p>
<h2>Noncompliant Code Example</h2>
<pre>
.mybox {
  color: ActiveBorder;
}
</pre>

<h2>See</h2>
<ul>
  <li><a href="http://dev.w3.org/csswg/css-color/#system-colors">W3C - CSS Color Module Level 4</a></li>
</ul>
