<p>Even though you can define any group of properties together in a CSS rule, some of them will be ignored due to the <code>display</code> of the element. This leads to extra cruft in your CSS and misunderstandings around how a rule should work.</p>
<p>For <code>display: inline</code>, the <code>width</code>, <code>height</code>, <code>margin-top</code>, <code>margin-bottom</code>, and <code>float</code> properties have no effect because inline elements don't have a formal box with which to apply the styles. The <code>margin-left</code> and <code>margin-right</code> properties still work reliably for indentation purposes but the other margin settings do not. The <code>float</code> property is sometimes used as a fix for the IE6 <a href="http://www.positioniseverything.net/explorer/doubled-margin.html">double-margin bug</a>.</p>
<p>Other general rules based on display are:
<ul>
  <li><code>display: inline-block</code> should not use <code>float</code>.</li>
  <li><code>display: block</code> should not use <code>vertical-align</code>.</li>
  <li><code>display: table-*</code> should not use <code>margin</code> (and all variants) or <code>float</code>.</li>
</ul>
</p>
<p>Removing the ignored or problematic properties decreases file size thereby improving performance.</p>
<h2>Noncompliant Code Example</h2>
<pre>
/* Noncompliant: 'display: inline' defined along with 'height' */
.mybox {
  display: inline;
  height: 25px;
}

/* Noncompliant: 'display: inline-block' defined along with 'float' */
.mybox {
  display: inline-block;
  float: left;
}

/* Noncompliant: 'display: table-cell' defined along with 'margin' */
.mybox {
  display: table-cell;
  margin: 10px;
}
</pre>
<h2>Compliant Code Example</h2>
<pre>
.mybox {
  display: inline;
  margin-left: 10px;
}

.mybox {
  display: table;
  margin-bottom: 10px;
}
</pre>
<h2>See</h2>
<ul>
  <li><a href="https://github.com/stubbornella/csslint/wiki/Require-properties-appropriate-for-display">CSSLint rule documentation</a></li>
</ul>
