<p>Even though vendor-specific properties are guaranteed not to cause conflicts, it should be recognized that these extensions may also be subject to change at the vendor’s whim, as they don’t form part of the CSS specifications, even though they often mimic the proposed behavior of existing or forthcoming CSS properties. Thus, it is not recommended to use them in production code.</p>
<h2>Noncompliant Code Example</h2>
<pre>
.mybox {
  -moz-border-radius: 5px;  /* Noncompliant */
  color: green;
}
</pre>
