<p>To make sure that your code will keep working as expected in the future, do not use obsolete properties (no longer supported by main vendors or support to be removed by main vendors).</p>
<h2>Noncompliant Code Example</h2>
<pre>
.mybox {
  azimuth: 30deg; /* Noncompliant */
  clip: rect(0px, 175px, 113px, 0px); /* Noncompliant */
}
</pre>
<h2>Obsolete Properties</h2>
<ul>
  <li>azimuth</li>
  <li>clip</li>
</ul>