<p>
    Sometimes when editing a rule you may end up defining multiple properties that can better be represented using
    shorthand. For example:
</p>
<pre>
.mybox {
  margin-left: 10px;
  margin-right: 10px;
  margin-top: 20px;
  margin-bottom: 30px;
}
</pre>
These four properties can actually be combined into a single margin property, such as:
<pre>
.mybox {
  margin: 20px 10px 30px;
}
</pre>
<p>
    Using shorthand properties where possible helps to decrease the overall size of the CSS.
</p>
<p>This rule is aimed at decreasing file size by finding properties that can be combined into one. As such, it raises an
    issue in the following cases:
</p>
<ul>
    <li>When <code>margin-left</code>, <code>margin-right</code>, <code>margin-top</code>, and
        <code>margin-bottom</code> are used together in a single rule.
    </li>
    <li>When <code>padding-left</code>, <code>padding-right</code>, <code>padding-top</code>, and
        <code>padding-bottom</code> are used together in a single rule.
    </li>
</ul>

<h2>Noncompliant Code Example</h2>
<pre>
.mybox {
  margin-left: 10px;
  margin-right: 10px;
  margin-top: 20px;
  margin-bottom: 30px;
}

.mybox {
  padding-left: 10px;
  padding-right: 10px;
  padding-top: 20px;
  padding-bottom: 30px;
}
</pre>

<h2>Compliant Solution</h2>
<pre>
/* Compliant: Shorthand magin property */
.mybox {
  margin: 20px 10px 30px;
}

/* Compliant: Shorthand padding property */
.mybox {
  padding: 20px 10px 30px;
}

/* Compliant: Only two margin properties */
.mybox {
  margin-left: 10px;
  margin-right: 10px;

}

/* Compliant: Only two padding properties */
.mybox {
  padding-right: 10px;
  padding-top: 20px;
}
</pre>

<h2>See</h2>
<ul>
    <li><a href="https://github.com/CSSLint/csslint/wiki/Require-shorthand-properties">CSSLint rule documentation</a>
    </li>
</ul>
