<p>The value of 0 works without specifying units in all situations where numbers with units or percentages are allowed. There is no difference between <code>0px</code>, <code>0em</code>, <code>0%</code>, or any other zero-value. The units aren't important because the value is still zero. CSS allows you to omit the units for zero values and still remain valid CSS.</p>
<p>It's recommended to remove units for all zero values because these units aren't being used by the browser and therefore can be safely removed to save bytes.</p>
<h2>Noncompliant Code Example</h2>
<pre>
.mybox {
  margin: 0px;
}

.mybox {
  width: 0%;
}

.mybox {
  padding: 10px 0px;
}
</pre>
<h2>Compliant Solution</h2>
<pre>
.mybox {
  margin: 0;
}

.mybox {
  width: 0;
}

.mybox {
  padding: 10px 0;
}
</pre>

<h2>See</h2>
<ul>
  <li><a href="https://github.com/stubbornella/csslint/wiki/Disallow-units-for-zero-values">CSSLint rule documentation</a></li>
</ul>
