<p>
    Prefer single-line comments (<code>//</code>) over multi-line (<code>/* ... */</code>) since they're not rendered in
    the final CSS.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
.mybox {
  /* blabla... */
  color: green;
}
</pre>

<h2>Compliant Solution</h2>
<pre>
.mybox {
  // blabla...
  color: green;
}
</pre>
