<p>For a better readability and understandability, Less variables should be defined at the beginning of their block.</p>
<p><code>@charset</code> and <code>@import</code> rules are allowed before Less variable declarations.</p>

<h2>Noncompliant Code Example</h2>
<pre>
@charset "UTF-8";

.mybox1 {
  color: @var1;
  @var1: blue; /* Noncompliant */
  @var2: green; /* Noncompliant */

  > .mybox2 {
    border-color: @var3;
    @var3: black; /* Noncompliant */
  }

  background-color: @var2;
  border: @var0;
}

@var0: 1px; /* Noncompliant */
</pre>

<h2>Compliant Solution</h2>
<pre>
@charset "UTF-8";

@var0: 1px;

.mybox1 {
  @var1: blue;
  @var2: green;
  color: @var1;

  > .mybox2 {
    @var3: black;
    border-color: @var3;
  }

  background-color: @var2;
  border: @var0;
}
</pre>
