<p>
    Early on in web development, including the same CSS property twice was certainly an error, especially if there were
    two different values. For example:
</p>
<pre>
.mybox {
  width: 100px;
  width: 120px;
}
</pre>
<p>
    Anyone looking at this code would think that this is clearly an error. Recently, however, including duplicate
    properties is used as a way to deal with varying levels of browser support for CSS properties. For example, some
    browsers support RGBA color while others do not, so it's quite common to see patterns such as:
</p>
<pre>
.mybox {
  background: #fff;
  background: rgba(255, 255, 255, 0.5);
}
</pre>
<p>
    This is quite clearly intentional. The developer wants to use RGBA when available but wants to fall back to a
    regular color when not available.
</p>
<p>This rule raises an issue when:</p>
<ul>
    <li>A property is included twice and contains the same value.</li>
    <li>A property is included twice and is separated by at least one other property.</li>
</ul>

<h2>Noncompliant Code Example</h2>
<pre>
/* Noncompliant: same value for the two declarations of the 'border' property */
.mybox {
  border: 1px solid black;
  border: 1px solid black;
}

/* Noncompliant: duplicated 'border' properties separated by another property */
.mybox {
  border: 1px solid black;
  color: green;
  border: 1px solid red;
}
</pre>

<h2>Compliant Solution</h2>
<pre>
/* Compliant: 'border' properties defined one after another with different values */
.mybox {
  border: 1px solid black;
  border: 1px solid red;
}
</pre>

<h2>stylelint Related Rules</h2>
<ul>
    <li><a href="https://stylelint.io/user-guide/rules/declaration-block-no-duplicate-properties/" target="_blank">declaration-block-no-duplicate-properties</a></li>
</ul>
