<p>Empty declarations are useless and should be removed to avoid confusing developers.</p>

<h2>Noncompliant Code Example</h2>
<pre>
.foo { ; }      /* Noncompliant */

.foo {
  ;             /* Noncompliant */
  color: green;
}
</pre>
