<p>
    An empty rule is one that doesn't contain any properties or variable declarations. A lot of times, empty rules
    appear as a result of refactoring without further cleanup. Eliminating empty rules results in smaller file sizes and
    less style information for the browser to deal with.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
.foo {
}

.foo {
  /* blabla */
}

@font-face {
}
</pre>

<h2>stylelint Related Rules</h2>
<ul>
    <li><a href="https://stylelint.io/user-guide/rules/block-no-empty/" target="_blank">block-no-empty</a></li>
</ul>
