<p>Empty declarations are useless and should be removed to avoid confusing developers.</p>

<h2>Noncompliant Code Example</h2>
<pre>
.foo { ; }      /* Noncompliant */

.foo {
  ;             /* Noncompliant */
  color: green;
}
</pre>

<h2>stylelint Related Rules</h2>
<ul>
  <li><a href="https://stylelint.io/user-guide/rules/no-extra-semicolons/" target="_blank">no-extra-semicolons</a></li>
</ul>
