<p>
    The <code>!important</code> flag must be placed at the end of the line, immediately before the semicolon,
    otherwise it will not be taken into account by the browser.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
.mybox {
  width: !important 100px;
  border: 1px solid !important black;
}
</pre>

<h2>Compliant Solution</h2>
<pre>
.mybox {
  width: 100px !important;
  border: 1px solid black !important;
}
</pre>

<h2>Exceptions</h2>
<p>
    For SCSS, the <code>!default</code> and <code>!global</code> flags are allowed after the <code>!important</code>
    flag. The following piece of code does not raise any issue:
</p>
<pre>
$var1: 10px !important !global;
$var2: 10px !important !default;
</pre>