<p>
    The <code>!important</code> annotation is used to artificially increase the specificity of a given property value in
    a rule. This is usually an indication that the specificity of the entire CSS has gotten a bit out of control and
    needs to be refactored. The more frequently <code>!important</code> is found, the more likely it is that
    developers are having trouble styling parts of a page effectively.
</p>

<h2>Noncompliant Code Example</h2>
<h3>CSS</h3>
<pre>
.mybox {
  color: red !important;
}
</pre>

<h3>Less</h3>
<pre>
.foo {
  .bar() !important;
}
</pre>

<h2>stylelint Related Rules</h2>
<ul>
    <li><a href="https://stylelint.io/user-guide/rules/declaration-no-important/" target="_blank">declaration-no-important</a></li>
    <li><a href="https://stylelint.io/user-guide/rules/keyframe-declaration-no-important/" target="_blank">keyframe-declaration-no-important</a></li>
</ul>
