<p>For readability reasons, remove the leading zeros of declaration values.</p>

<h2>Noncompliant Code Example</h2>
<pre>
h1 {
 font-size: 0.9em;
}
</pre>

<h2>Compliant Solution</h2>
<pre>
h1 {
 font-size: .9em;
}
</pre>

<h2>stylelint Related Rules</h2>
<ul>
 <li><a href="https://stylelint.io/user-guide/rules/number-leading-zero/" target="_blank">number-leading-zero</a></li>
</ul>
