<p>For readability reasons, remove the leading zeros of declaration values.</p>

<h2>Noncompliant Code Example</h2>
<pre>
h1 {
 font-size: 0.9em;
}
</pre>

<h2>Compliant Solution</h2>
<pre>
h1 {
 font-size: .9em;
}
</pre>
