<p>For readability reasons, each declaration should be defined on a separate line.</p>

<h2>Noncompliant Code Example</h2>
<pre>
.mybox {
  background-color: blue;
  color: red; border-color: green;
}
</pre>

<h2>Compliant Solution</h2>
<pre>
.mybox {
  background-color: blue;
  color: red;
  border-color: green;
}
</pre>

<h2>stylelint Related Rules</h2>
<ul>
  <li><a href="https://stylelint.io/user-guide/rules/block-no-single-line/" target="_blank">block-no-single-line</a></li>
  <li><a href="https://stylelint.io/user-guide/rules/declaration-block-single-line-max-declarations/" target="_blank">declaration-block-single-line-max-declarations</a></li>
</ul>
