<p>
    Avoid using too specific selectors. This means that when you have any style you want that is an exception from the
    common styling, it will be easier to implement it and override the default applied styling.
</p>

<h2>Noncompliant Code Example</h2>
With a maximum depth of 3:
<pre>
div#container div#header ul#navigation li{
  float: left;
}
</pre>

<h2>Compliant Code Example</h2>
With a maximum depth of 3:
<pre>
#navigation li{
  float: left;
}
</pre>

<h2>See</h2>
<ul>
    <li><a href="http://robertnyman.com/2007/10/18/dont-over-specify-your-css-code/">Don't Over-Specify Your CSS
        Code</a></li>
</ul>

<h2>stylelint Related Rules</h2>
<ul>
    <li><a href="https://stylelint.io/user-guide/rules/selector-max-compound-selectors/" target="_blank">selector-max-compound-selectors</a></li>
</ul>
