<p> For security reasons, protocol-relative URLs should not be used.</p>

<h2>Noncompliant Code Example</h2>
<pre>
@font-face {
  src: url("//myfont");
}
</pre>

<h2>Compliant Solution</h2>
<pre>
@font-face {
  src: url("https://myfont");
}
</pre>

<h2>See</h2>
<ul>
    <li><a target="_blank" href="https://www.paulirish.com/2010/the-protocol-relative-url/">Paul Irish blog post</a>
    </li>
</ul>

<h2>stylelint Related Rules</h2>
<ul>
    <li><a href="https://stylelint.io/user-guide/rules/function-url-no-scheme-relative/" target="_blank">function-url-no-scheme-relative</a></li>
</ul>
