<p> All URLs should be enclosed in quotes as enforced by the most recent W3C specifications.</p>

<h2>Noncompliant Code Example</h2>
<pre>
@font-face {
  src: url(myfont);
}
</pre>

<h2>Compliant Solution</h2>
<pre>
@font-face {
  src: url('myfont');
}
</pre>

<h2>See</h2>
<ul>
    <li><a target="_blank" href="https://drafts.csswg.org/css-values-3/#urls">CSS Values and Units Module Level 3 W3C Specifications</a>
    </li>
</ul>

<h2>stylelint Related Rules</h2>
<ul>
    <li><a href="https://stylelint.io/user-guide/rules/function-url-quotes/" target="_blank">function-url-quotes</a></li>
</ul>
