<p>This rule is deprecated. It is now split into three dedicated rules:</p>
<ul>
    <li>Class selectors should follow a naming convention</li>
    <li>ID selectors should follow a naming convention</li>
    <li>Placeholder selectors should follow a naming convention</li>
</ul>
<p>
    Shared coding conventions allow teams to collaborate efficiently. This rule checks that all selector names match a
    provided regular expression.
</p>
<p>
    The default regular expression only allows lowercase letters, digits and hyphens. Uppercase letters are not allowed
    because CSS selectors are used in HTML. As HTML is case-insensitive, it is usually written in lowercase and it may
    be confusing and even error-prone to add CSS selectors with uppercase letters to the HTML code.
</p>

<h2>Noncompliant Code Example</h2>
With format <code>^[a-z][-a-z0-9]*$</code>:
<pre>
.myBox {  /* Noncompliant: Should be 'mybox' or 'my-box' instead for example */
}

.my_box {  /* Noncompliant: Should be 'mybox' or 'my-box' instead for example */
}
</pre>

<h2>Compliant Solution</h2>
With format <code>^[a-z][-a-z0-9]*$</code>:
<pre>
.mybox {
}

.my-box {
}
</pre>

<h2>Exceptions</h2>
<p>
    Less interpolated selectors are not checked for naming convention. For instance, the following piece of code does
    not raise an issue:
</p>
<pre>
.abc-@{DEF} {
  color: green;
}
</pre>

<p>
    SCSS interpolated selectors are not checked for naming convention. For instance, the following piece of code does
    not raise an issue:
</p>
<pre>
.abc-#{$DEF} {
  color: green;
}
</pre>
