<p>For readability reasons, remove the trailing zeros for numeric values.</p>

<h2>Noncompliant Code Example</h2>
<pre>
h1 {
 font-size: 12.0px;
}
</pre>

<h2>Compliant Solution</h2>
<pre>
h1 {
 font-size: 12px;
}
</pre>

<h2>stylelint Related Rules</h2>
<ul>
 <li><a href="https://stylelint.io/user-guide/rules/number-no-trailing-zeros/" target="_blank">number-no-trailing-zeros</a></li>
</ul>
