<p>For readability reasons, remove the trailing zeros for numeric values.</p>

<h2>Noncompliant Code Example</h2>
<pre>
h1 {
 font-size: 12.0px;
}
</pre>

<h2>Compliant Solution</h2>
<pre>
h1 {
 font-size: 12px;
}
</pre>
