<p>
    The universal selector (<code>*</code>) matches all elements. Though convenient for selecting a group of elements at
    a time, the universal selector causes performance issues when used as the key part (far-right) of a selector.
</p>
<p>
    Browsers evaluate selectors from right-to-left, so the follwing rule begins by first matching every element in the
    document.
    After that, each of those elements must be inspected to see if it matches the next criteria, which is having an
    ancestor with the class of <code>mybox</code>. The more complex the selector containing <code>*</code>, the slower
    the evaluation
    becomes. For this reason, it's recommended to avoid using the universal selector as the key part of a selector.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
.mybox * {
  background: #fff;
  color: #000;
  background: rgba(255, 255, 255, 0.5);
}

* {
  color: red;
}
</pre>

<h2>Compliant Solution</h2>
<pre>
.selected * a {
  color: red;
}
</pre>

<h2>stylelint Related Rules</h2>
<ul>
    <li><a href="https://stylelint.io/user-guide/rules/selector-no-universal/" target="_blank">selector-no-universal</a></li>
</ul>
