<p>
    The value of 0 works without specifying unit for length and percentage. There is no difference between
    <code>0px</code>, <code>0em</code>, <code>0%</code>, or any other zero-value. The units are not important because
    the value is still zero. CSS allows you to omit the units for zero values and still remain valid CSS.
</p>
<p>
    It's recommended to remove units for all zero values because these units aren't being used by the browser and
    therefore can be safely removed to save bytes.
</p>
<p>
    Note that it is applicable for length and percentage but not for other dimensions such as angle, time, etc.
</p>
<h2>Noncompliant Code Example</h2>
<pre>
.mybox {
  margin: 0px;
  width: 0%;
  padding: 10px 0px;
}
</pre>
<h2>Compliant Solution</h2>
<pre>
.mybox {
  margin: 0;
  width: 0;
  padding: 10px 0;
}

.mybox {
  angle: 0grad;
  transition-duration: 0s;
}
</pre>

<h2>See</h2>
<ul>
    <li><a href="https://drafts.csswg.org/css-values-3/#lengths">W3C Specifications</a></li>
    <li><a href="https://developer.mozilla.org/en/docs/Web/CSS/angle#Summary">MDN CSS Reference &lt;angle&gt;></a></li>
</ul>
