<p>
    Experimental CSS properties are typically implemented using vendor prefixes until the final behavior has been
    established and agreed upon. Most CSS3 properties have vendor-prefixed equivalents for multiple vendors, including
    Firefox (<code>-moz-</code>), Safari/Chrome (<code>-webkit-</code>) and Internet Explorer (<code>-ms-</code>). It's
    easy to forget to include the vendor prefixed version of a property when there are so many
    to keep track of.
</p>
<p>
    This rule is intended to raise an issue when a vendor-prefixed property is missing. Note that only the following
    main vendors are checked by this rule:
</p>
<ul>
    <li>Firefox (<code>-moz-</code>)</li>
    <li>Safari/Chrome (<code>-webkit-</code>)</li>
    <li>Internet Explorer (<code>-ms-</code>)</li>
</ul>
<p>Opera (<code>-o-</code>) for example is not.</p>
<p>The following properties have multiple vendor-prefixed versions:</p>
<table border=1>
    <thead>
    <tr align=center>
        <th>Property</th>
        <th>Vendor Prefixes</th>
    </tr>
    </thead>
    [[vendorPrefixedProperties]]
</table>

<h2>Noncompliant Code Example</h2>
<pre>
/* Noncompliant: Missing -ms- */
.mybox {
  -webkit-hyphens: none;
  hyphens: none;
}
</pre>

<h2>Compliant Solution</h2>
<pre>
.mybox {
  -ms-hyphens: none;
  -webkit-hyphens: none;
  hyphens: none;
}
</pre>
