<p>
    Experimental CSS properties are typically implemented using vendor prefixes until the final behavior has been
    established and agreed upon. Most CSS3 properties have vendor-prefixed equivalents for multiple vendors, including
    Firefox (<code>-moz-</code>), Safari/Chrome (<code>-webkit-</code>), Opera (<code>-o-</code>) and Internet Explorer
    (<code>-ms-</code>). It's easy to forget to include the vendor prefixed version of a property when there are so many
    to keep track of.
</p>
<p>The following properties have multiple vendor-prefixed versions:</p>
[[vendorPrefixedProperties]]

<p>
    This rule is intended to raise an issue when a vendor-prefixed property is missing. Note that only the following
    main vendors are checked by this rule:
</p>
<ul>
    <li>Firefox (<code>-moz-</code>)</li>
    <li>Safari/Chrome (<code>-webkit-</code>)</li>
    <li>Opera (<code>-o-</code>)</li>
    <li>Internet Explorer (<code>-ms-</code>)</li>
</ul>
<p>Konqueror (<code>-khtml-</code>) for example is not.</p>

<h2>Noncompliant Code Example</h2>
<pre>
/* Noncompliant: Missing -moz-, -ms-, and -o- */
.mybox {
  -webkit-transform: translate(50px, 100px);
}
</pre>

<h2>Compliant Solution</h2>
<pre>
.mybox {
  -webkit-transform: translate(50px, 100px);
  -moz-transform: translate(50px, 100px);
  -ms-transform: translate(50px, 100px);
  -o-transform: translate(50px, 100px);
}
</pre>
