<p>
    Even though vendor-specific functions are guaranteed not to cause conflicts, it should be recognized that these
    extensions may also be subject to change at the vendor’s whim, as they don’t form part of the CSS specifications,
    even though they often mimic the proposed behavior of existing or forthcoming CSS functions. Thus, it is not
    recommended to use them in production code.
</p>

<p>The rule raises an issue when one of the following prefixes is found:</p>
[[vendors]]

<p>This rule also raises an issue when one of the following experimental functions is found:</p>
[[experimentalFunctions]]

<h2>Noncompliant Code Example</h2>
<pre>
.mybox {
  background-image: -moz-linear-gradient(top, #D7D 0%, #068 100%); /* Noncompliant: vendor-prefixed function */
  background: #0ac repeating-conic-gradient(at 20%, white 0deg, white 20deg, red 20deg, red 40deg); /* Noncompliant: experimental function */
}
</pre>
