<p>To make sure that your code will keep working as expected in the future, do not use obsolete properties:</p>
<ul>
    <li>No longer supported by main vendors
    <li>Support to be removed by main vendors</li>
</ul>

<h2>Noncompliant Code Example</h2>
<pre>
.mybox {
  azimuth: 30deg; /* Noncompliant */
  appearance: normal; /* Noncompliant */
}
</pre>

<h2>Obsolete Properties</h2>
[obsoleteProperties]
