<p>To make sure that your code will keep working as expected in the future, do not use obsolete peudo-elements and pseudo-classes:</p>
<ul>
    <li>No longer supported by main vendors
    <li>Support to be removed by main vendors</li>
</ul>

<h2>Noncompliant Code Example</h2>
<pre>
:any(ol, ul, menu, dir) {
  list-style-type: square;
}
</pre>

<h2>Obsolete Pseudo-Elements and Pseudo-Classes</h2>
[[obsoletePseudos]]
