<p>
    Browsers ignore invalid property values. Review those invalid property values and either remove them or update them.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
/* 'word-spacing' validator is 'normal | &lt;length&gt;' */
.mybox {
  word-spacing: 10px;
  word-spacing: normal;
  word-spacing: 0;
  word-spacing: inherit;
  word-spacing: 10;       /* Noncompliant: Missing length unit */
  word-spacing: abc;      /* Noncompliant: 'abc' value is not allowed */
}
</pre>

<h2>Implemented Validators</h2>
<p>
    Based on <a target="_blank" href="http://www.w3.org/Style/CSS/current-work">W3C CSS specifications</a> Level 1 or 2
    or 3 which status is either Completed or Testing or Refining. Note that not all validators are implemented for now.
</p>
