<p>
    Browsers ignore invalid property values. Review those invalid property values and either remove them or update them.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
/* 'word-spacing' validator is 'normal | &lt;length&gt; | &lt;percentage&gt;' */
.mybox {
  word-spacing: 10px;
  word-spacing: normal;
  word-spacing: 0;
  word-spacing: inherit;
  word-spacing: 10%;
  word-spacing: 10;       /* Noncompliant: Missing length unit */
  word-spacing: abc;      /* Noncompliant: 'abc' value is not allowed */
}
</pre>

<h2>Implemented Validators</h2>
<p>Based on <a target="_blank" href="http://www.w3.org/Style/CSS/current-work">W3C CSS specifications</a>.</p>
<p>Status</p>
<ul>
    <li>Supported: Supported by all main vendors (Firefox, Chrome, IE/Edge)</li>
    <li>Experimental: Not supported by all main vendors (Firefox, Chrome, IE/Edge)</li>
</ul>
<table border=1>
    <thead>
    <tr align=center>
        <th>Property</th>
        <th>Validator</th>
        <th>Status</th>
    </tr>
    </thead>
    [[propertyValidators]]
</table>

