<p>To make sure that your code will keep working as expected in the future, do not use obsolete functions:</p>
<ul>
    <li>No longer supported by main vendors
    <li>Support to be removed by main vendors</li>
</ul>

<h2>Noncompliant Code Example</h2>
<pre>
.mybox {
  font-size: min(10px, 3em); /* Noncompliant */
  font-size: max(10px, 3em); /* Noncompliant */
}
</pre>

<h2>Obsolete Functions</h2>
[[obsoleteNotLessFunctions]]
