<p>
    Even though CSS syntax is case-insensitive (except for variables), sharing some style conventions is a key point to
    make it possible for a team to efficiently collaborate.
</p>
<p> This rule checks that all the following elements are written in lower case: </p>
<ul>
    <li>properties</li>
    <li>functions</li>
    <li>@-rule keywords</li>
    <li>type selectors</li>
    <li>pseudo functions</li>
    <li>pseudo identifiers</li>
    <li>units</li>
    <li>hexadecimal values</li>
</ul>

<h2>Noncompliant Code Example</h2>
<pre>
@CHARSET "UTF-8"; /* Noncompliant CHARSET @-rule keyword */

.mybox {
  COLOR: blue; /* Noncompliant COLOR property*/
}

.mybox {
  color: RGB(255,255,255); /* Noncompliant RGB function */
}

A:hover {} /* Noncompliant A type selector */

a:HOVER {} /* Noncompliant HOVER pseudo identifier */

*:NOT(foo) {} /* Noncompliant NOT pseudo function */

.mybox {
  width: 10PX; /* Noncompliant PX unit */
  color: #FFFFFF; /* Noncompliant #FFFFFF hexadecimal value */
}
</pre>

<h2>Compliant Solution</h2>
<pre>
@charset "UTF-8";

.mybox {
  color: blue;
}

.mybox {
  color: rgb(255,255,255);
}

a:hover {}

*:not(foo) {}


.mybox {
  width: 10px;
  color: #ffffff;
}
</pre>

<h2>Exceptions</h2>
[begin-less]
<p>
    Less interpolated properties are not checked for lowercase. For instance, the following piece of code does not raise
    any issue:
</p>
<pre>
.mybox {
  @{BG}-color: green;
}
</pre>
[end-less]

[begin-scss]
<p>
    SCSS interpolated properties are not checked for lowercase. For instance, the following piece of code does not raise
    any issue:
</p>
<pre>
.mybox {
  #{$BG}-color: green;
}
</pre>
[end-scss]

<p>
    The following functions can be either written in lowercase or as follow:
</p>
[[functionCaseExceptions]]

<h2>stylelint Related Rules</h2>
<ul>
    <li><a href="https://stylelint.io/user-guide/rules/property-case/" target="_blank">property-case</a></li>
    <li><a href="https://stylelint.io/user-guide/rules/function-name-case/" target="_blank">function-name-case</a></li>
    <li><a href="https://stylelint.io/user-guide/rules/at-rule-name-case/" target="_blank">at-rule-name-case</a></li>
    <li><a href="https://stylelint.io/user-guide/rules/selector-pseudo-class-case/" target="_blank">selector-pseudo-class-case</a>
    </li>
    <li><a href="https://stylelint.io/user-guide/rules/selector-pseudo-element-case/" target="_blank">selector-pseudo-element-case</a>
    </li>
    <li><a href="https://stylelint.io/user-guide/rules/selector-type-case/" target="_blank">selector-type-case</a></li>
    <li><a href="https://stylelint.io/user-guide/rules/unit-case/" target="_blank">unit-case</a></li>
    <li><a href="https://stylelint.io/user-guide/rules/color-hex-case/" target="_blank">color-hex-case</a></li>
</ul>
