<p>
    Even though CSS syntax is case-insensitive (except for variables), sharing some style conventions is a key point to
    make it possible for a team to efficiently collaborate.
</p>
<p>
    This rule checks that all properties, functions, @-rule keywords, pseudo functions and pseudo identifiers are
    written in lowercase.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
@CHARSET "UTF-8"; /* Noncompliant CHARSET @-rule keyword */

.mybox {
  COLOR: blue; /* Noncompliant COLOR property*/
}

.mybox {
  color: RGB(255,255,255); /* Noncompliant RGB function */
}

a:HOVER {} /* Noncompliant HOVER pseudo identifier */

*:NOT(foo) {} /* Noncompliant NOT pseudo function */
</pre>

<h2>Compliant Solution</h2>
<pre>
@charset "UTF-8";

.mybox {
  color: blue;
}

.mybox {
  color: rgb(255,255,255);
}

a:hover {}

*:not(foo) {}
</pre>

<h2>Exceptions</h2>
<p>
    Less interpolated properties are not checked for lowercase. For instance, the following piece of code does not raise
    any issue:
</p>
<pre>
.mybox {
  @{BG}-color: green;
}
</pre>

<p>
    SCSS interpolated properties are not checked for lowercase. For instance, the following piece of code does not raise
    any issue:
</p>
<pre>
.mybox {
  #{$BG}-color: green;
}
</pre>

<p>
    The following functions can be either written in lowercase or as follow:
</p>
[[functionCaseExceptions]]
