<p>
    Shared coding conventions allow teams to collaborate efficiently. This rule checks that all class selector names
    match a provided regular expression.
</p>
<p>
    The default regular expression only allows lowercase letters, digits and hyphens. Uppercase letters are not allowed
    because CSS selectors are used in HTML. As HTML is case-insensitive, it is usually written in lowercase and it may
    be confusing and even error-prone to add CSS selectors with uppercase letters to the HTML code.
</p>
<p>
    The regular expression can be customized depending on your naming conventions. For example, if you follow <a
        href="http://csswizardry.com/2015/03/more-transparent-ui-code-with-namespaces" target="_blank">BEM naming
    conventions</a>, set the regular expression to
    <code>^(_)?[a-z]+-[a-z0-9-]+((_{2}|-{2})?[a-z0-9-]+)?(-{2}[a-z0-9-]+)?[a-z0-9]$</code>.
</p>

<h2>Noncompliant Code Example</h2>
With format <code>^[a-z][-a-z0-9]*$</code>:
<pre>
.myBox {  /* Noncompliant: Should be 'mybox' or 'my-box' instead for example */
  color: green;
}

.my_box {  /* Noncompliant: Should be 'mybox' or 'my-box' instead for example */
  color: green;
}
</pre>

<h2>Compliant Solution</h2>
With format <code>^[a-z][-a-z0-9]*$</code>:
<pre>
.mybox {
  color: green;
}

.my-box {
  color: green;
}
</pre>

[begin-less]
<h2>Exceptions</h2>
<p>
    Less interpolated class selectors are not checked for naming convention. For instance, the following piece of code
    does not raise an issue:
</p>
<pre>
.abc-@{DEF} {
  color: green;
}
</pre>
[end-less]

[begin-scss]
<h2>Exceptions</h2>
<p>
    SCSS interpolated class selectors are not checked for naming convention. For instance, the following piece of code
    does not raise an issue:
</p>
<pre>
.abc-#{$DEF} {
  color: green;
}
</pre>
[end-scss]

<h2>stylelint Related Rules</h2>
<ul>
    <li><a href="https://stylelint.io/user-guide/rules/selector-class-pattern/" target="_blank">selector-class-pattern</a></li>
</ul>
