<p>
    Even though vendor-specific identifiers are guaranteed not to cause conflicts, it should be recognized that these
    extensions may also be subject to change at the vendor’s whim, as they don’t form part of the CSS specifications,
    even though they often mimic the proposed behavior of existing or forthcoming CSS identifiers. Thus, it is not
    recommended to use them in production code.
</p>

<p>The rule raises an issue when one of the following prefixes is found:</p>
[[vendors]]

<h2>Noncompliant Code Example</h2>
<pre>
.mybox {
  display: flex;
  display: -ms-flexbox; /* Noncompliant */
}
</pre>

<h2>stylelint Related Rules</h2>
<ul>
    <li><a href="https://stylelint.io/user-guide/rules/value-no-vendor-prefix/" target="_blank">value-no-vendor-prefix</a></li>
</ul>
