<p>
    Even though vendor-specific properties are guaranteed not to cause conflicts, it should be recognized that these
    extensions may also be subject to change at the vendor’s whim, as they don’t form part of the CSS specifications,
    even though they often mimic the proposed behavior of existing or forthcoming CSS properties. Thus, it is not
    recommended to use them in production code.
</p>

<p>The rule raises an issue when one of the following prefixes is found:</p>
[[vendors]]

<p>This rule also raises an issue when one of the following experimental properties is found:</p>
[[experimentalProperties]]

<h2>Noncompliant Code Example</h2>
<pre>
.mybox {
  -moz-border-radius: 5px;  /* Noncompliant: vendor-prefixed property */
  pause: 3s;  /* Noncompliant: experimental property */
}
</pre>
