<p>
    Sometimes when editing a rule you may end up defining multiple properties that can better be represented using
    shorthand. For example:
</p>
<pre>
.mybox {
  margin-left: 10px;
  margin-right: 10px;
  margin-top: 20px;
  margin-bottom: 30px;
}
</pre>
These four properties can actually be combined into a single <code>margin</code> property, such as:
<pre>
.mybox {
  margin: 20px 10px 30px;
}
</pre>
<p>This rule helps decrease file size by finding properties that can be combined into one.</p>

<h2>Noncompliant Code Example</h2>
<pre>
.mybox {
  margin-left: 10px;
  margin-right: 10px;
  margin-top: 20px;
  margin-bottom: 30px;
}

.mybox {
  padding-left: 10px;
  padding-right: 10px;
  padding-top: 20px;
  padding-bottom: 30px;
}
</pre>

<h2>Compliant Solution</h2>
<pre>
/* Compliant: Shorthand magin property */
.mybox {
  margin: 20px 10px 30px;
}

/* Compliant: Shorthand padding property */
.mybox {
  padding: 20px 10px 30px;
}

/* Compliant: Only two margin properties */
.mybox {
  margin-left: 10px;
  margin-right: 10px;

}

/* Compliant: Only two padding properties */
.mybox {
  padding-right: 10px;
  padding-top: 20px;
}
</pre>

<h2>See</h2>
<ul>
    <li>
        Related rule: {rule:css:do-not-use-shorthand-properties}
    </li>
</ul>

<h2>Shorthand Properties</h2>
<table border=1>
    <thead>
    <tr align=center>
        <th>Property</th>
        <th>Shorthand For</th>
    </tr>
    </thead>
    [[shorthandProperties]]
</table>

<h2>stylelint Related Rules</h2>
<ul>
    <li><a href="https://stylelint.io/user-guide/rules/declaration-block-no-redundant-longhand-properties/" target="_blank">declaration-block-no-redundant-longhand-properties</a></li>
</ul>
