<p>
    Shared coding conventions allow teams to collaborate efficiently. This rule checks that all CSS variable names
    match a provided regular expression.
</p>

<h2>Noncompliant Code Example</h2>
With format <code>^[a-z][-a-z0-9]*$</code>:
<pre>
--myColor: green; /* Noncompliant */
--my_color: green; /* Noncompliant */
</pre>

<h2>Compliant Solution</h2>
With format <code>^[a-z][-a-z0-9]*$</code>:
<pre>
--my-color: green;
</pre>

<h2>stylelint Related Rules</h2>
<ul>
    <li><a href="https://stylelint.io/user-guide/rules/custom-property-pattern/" target="_blank">custom-property-pattern</a></li>
</ul>
